src/components/interface/AnalysisFooter.tsx

```diff
@@ -65,6 +64,53 @@ export function AnalysisFooter() {
 
   const isStart = useMemo(() => currentStep === 0, [currentStep]);
   const isEnd = useMemo(() => currentStep === flatSequence.length, [currentStep, flatSequence.length]);
+  const identifier = useCurrentIdentifier();
+  const [hasAudio, setHasAudio] = useState(false);
+
+  useEffect(() => {
+    if (!storageEngine || !participantId || !identifier) {
+      setHasAudio(false);
+      return;
+    }
+
+    storageEngine.getAudioUrl(identifier, participantId)
+      .then((audioUrl) => setHasAudio(!!audioUrl))
+      .catch(() => setHasAudio(false));
+  }, [storageEngine, participantId, identifier]);
+
+  const handleDownloadAudio = async () => {
+    if (!storageEngine || !participantId || !identifier) {
+      return;
+    }
+
+    const audioUrl = await storageEngine.getAudioUrl(identifier, participantId);
```

Comment:
I think you can use a useState here and avoid pulling the audio url multiple times. Maybe instead of hasAudio, you can just make audioUrl in to the state and handle that